crates/autopilot/src/domain/auction/order.rs

```diff
@@ -1,6 +1,7 @@
 use {
     crate::domain::{eth, fee},
     primitive_types::{H160, H256, U256},
+    shared::app_data::ValidatedAppData,
```

Comment:
I'm not a fan of making the autopilot domain depend on the `shared` crate. The shared crate is one of the harder to remove dependencies (as it contains a lot of random code we put in there in the beginning).

I wonder if it would be easy to move the `app_data` module out of the shared crate into its own module (from briefly checking it would also have to contain the `Hooks` type).

To not block this PR can we at least hide this dependency behind a boundary module and have the autopilot domain (in particular the `fee` module) reference it via `crate::boundary`?